.expeditor/scripts/website-deploy.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# This deploys the web site
+
+set -euo pipefail
+
+# Explicitly get the current directory rather than assuming this is being run
+# from the root of the habitat project, because this script is called from
+# multiple places.
+mydir="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
```

Comment:
Yeah, agreed. This tends to make additional script libraries a bit hairy, in my experience thus far.

Reply:
Here's my goal:  There should be exactly 1 way to deploy the website (e.g. 1 command to run), whether that deploy is happening via CI or whether it's happening manually.

You've probably noticed that the work here only deploys the production site, not acceptance.  Initially I wanted to deploy the acceptance web site whenever a PR was opened, but after some discussion with @tduffield , it was decided that was a potential security hole.

But, we still need to do acceptance deploys.  That means someone is going to have to do the thing to deploy acceptance manually.  I want 1 command that anyone can run and deploy the web site, and I want CI to use the exact same script/command.

That was my goal with this line, and with this PR in general.  If there's a better way to accomplish it, I'm happy to change it, but I don't want to assume that it will always be run in CI.  That defeats the goal.

Reply:
To get a little subtle, it's not so much that this script is intended to be run by CI, _in_ CI, but that this script (and others in the `.expeditor` directory) are broadly assumed to be _invoked_ from the root of the repository.

This script could do that, and the `deploy_XXX` Makefile targets in the `www` directory could be changed to something like this:

```
deploy_acceptance:
    cd ..; .expeditor/scripts/website-deploy.sh
```

Since things in the `.expeditor` directory are generally assumed to be called from the root, it would be clear why those makefile targets look a little different. By removing the `BASH_SOURCE[0]` code from the script now, we'd also keep the script in line with others in that directory, and remove the temptation to refactor later down the line, perhaps inadvertently breaking our manual deploy in the process.

Reply:
Sure, sounds good

Reply:
Fixed in https://github.com/habitat-sh/habitat/pull/7273/commits/7eecaa4a2b60b1610076b0c703b06df1cdd2ec6c